### PR TITLE
[dhctl] Bootstrap deckhouse with a version tag instead of release channel tag and refactor preflight checks messages

### LINKED ISSUE #3251: [dhctl] image updated to v* tag after bootstrap ends
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

main

### Expected Behavior

I assume that deckhouse is ready after issueing this command:

```
dhctl bootstrap-phase install-deckhouse --kubeconfig=/kubeconfig --config=/config.yml
```
(It is from the [Getting started for kind](https://deckhouse.io/gs/kind/step4.html))

### Actual Behavior

`dhctl` says deckhouse is ready:

```
...

│ │ │ Deckhouse pod found: deckhouse-fb74f4595-lvsf8 (Running)
│ │ │ Running pod found! Checking logs...
│ │ │ 	Module "operator-prometheus-crd" run successfully
│ │ │ 	Module "prometheus" run successfully
│ │ │ 	Module "ingress-nginx" run successfully
│ │ │ No more converge tasks found in Deckhouse queue.
│ │ │ Deckhouse pod is Ready!
│ │ └ Waiting for Deckhouse to become Ready (108.88 seconds)
│ └ ⛵ ~ Bootstrap: Install Deckhouse (109.08 seconds)
└ ⛵ ~ Bootstrap: Install Deckhouse (109.18 seconds)
```

But it is not ready:

```
kubectl -n d8-system get po
NAME                                        READY   STATUS    RESTARTS       AGE
deckhouse-5b997f9d64-qkfnx                  0/1     Running   0              70s
```

It updates image from `ce:stable` to `ce:v1.40.3` (`ce:alpha -> ce:v1.41.1`) and starts pulling:

  Normal   Pulling    3m12s             kubelet            Pulling image "registry.deckhouse.io/deckhouse/ce:v1.40.3"
  Normal   Pulled     20s               kubelet            Successfully pulled image "registry.deckhouse.io/deckhouse/ce:v1.40.3" in 2m51.6355445s

### Steps To Reproduce

1. Install deckhouse in kind: https://deckhouse.io/gs/kind/step4.html
2. Run dhctl bootstrap-phase install-deckhouse
3. See message "Deckhouse pod is Ready!"
4. Create ingress controller `kubectl create -f ingress-nginx-controller.yml`
5. Be confused by the error:
```
error: unable to recognize "ingress-controller.yml": no matches for kind "IngressNginxController" in version "deckhouse.io/v1"
```

### PR DESCRIPTION
**Pull request by mvasl**
## Description

Adjusted image tag selection logic to allow usage of deckhouse version tag from `/deckhouse/version` for installation. Deckhouse is now bootstrapped using this tag.

Also preflight checks code was refactored.

And refactored preflight check error messages.

## Why do we need it, and what problem does it solve?

Fixes #3251 

image

image

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl
type: feature
summary: Allow installing Deckhouse from tag. Refactor preflight checks code.
impact_level: default
```

### CODE REVIEW

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
allmost all ok

File: dhctl/pkg/kubernetes/actions/deckhouse/install.go [hunk 1]
```diff
@@ -59,19 +62,42 @@ type Config struct {
...
+	switch {
+	case errors.Is(err, fs.ErrNotExist):
+		return "", false
+	case err != nil:
+		panic(err)
+	}
+
+	if !bytes.HasPrefix(rawFile, []byte("v")) && !bytes.Contains(rawFile, []byte(".")) {
+		return "", false
+	}
```
Comment (RomanenkoDenys) [inline dhctl/pkg/kubernetes/actions/deckhouse/install.go:96]: please use https://pkg.go.dev/golang.org/x/mod/semver#IsValid here

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
allmost all ok

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

File: dhctl/pkg/preflight/dhctl_obsolescence.go [hunk 2]
```diff
@@ -123,7 +123,7 @@ func (pc *PreflightCheck) fetchDeckhouseImageHashFromReleaseChannel(ctx context.
 		return nil, fmt.Errorf("parse ClusterConfiguration.deckhouse.registryDockerCfg: %w", err)
 	}
 
-	imageReference, err := name.ParseReference(pc.installConfig.GetImage())
```
Comment (name212) [inline dhctl/pkg/preflight/dhctl_obsolescence.go:126]: why are not we using for preflight check force using tag?

Comment (name212) [reply dhctl/pkg/preflight/dhctl_obsolescence.go:126]: Why compare with the image of the release channel? When we insert an end tag into the deployment. If the final goal is that no updates occur during installation, then we need to compare both the image tag for the tag and for the release channel, if there is one. In general, the release channel planned to transfer the config from the init configuration to the config module and remove it from the init configuration.

Comment (name212) [reply dhctl/pkg/preflight/dhctl_obsolescence.go:126]: >to transfer the config from the init configuration to the config module and remove it from the init configuration.
and then it becomes optional altogether

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/static'...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.25 6347886355 6347886355-1-con-1-25 /sys/deckhouse-oss/install:pr5955 astra@192.168.199.154

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by name212 — APPROVED** [review]

File: dhctl/pkg/preflight/dhctl_obsolescence.go [hunk 3]
```diff
@@ -113,30 +113,43 @@ func (pc *PreflightCheck) CheckDhctlVersionObsolescence() error {
...
+	channelRef, err := name.ParseReference(pc.installConfig.GetImage(false))
 	if err != nil {
 		return nil, fmt.Errorf("parse image refernce: %w", err)
 	}
 
-	descriptor, err := pc.imageDescriptorProvider.Descriptor(imageReference, remote.WithContext(ctx), remote.WithAuth(creds))
+	versionTagRef, err := name.ParseReference(pc.installConfig.GetImage(true))
+	if err != nil {
+		return nil, fmt.Errorf("parse image refernce: %w", err)
```
Comment (RomanenkoDenys) [inline dhctl/pkg/preflight/dhctl_obsolescence.go:132]:
```suggestion
		return nil, fmt.Errorf("parse image reference: %w", err)
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
one mistake

**Review by name212 — DISMISSED** [review]

**Review by name212 — APPROVED** [review]

**Pull request merged**